spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/post/SendResponseFilter.java

```diff
@@ -56,6 +53,17 @@
 	private static DynamicBooleanProperty SET_CONTENT_LENGTH = DynamicPropertyFactory
 			.getInstance()
 			.getBooleanProperty(ZuulConstants.ZUUL_SET_CONTENT_LENGTH, false);
+	private boolean useServlet31 = true;
+
+	public SendResponseFilter() {
+		super();
+		// To support Servlet API 3.0.1 we need to check if setcontentLengthLong exists
+		try {
+			HttpServletResponse.class.getMethod("setContentLengthLong");
+		} catch(NoSuchMethodException e) {
+			useServlet31 = false;
```

Comment:
We face an issue that useServlet31 is always false. 
In that case if the content-lenght is bigger that max int there is not content-length value in the response headers. 
HttpServletResponse.class.getMethod("setContentLengthLong"); gives always a NoSuchMethodException, as there is not setContentLengthLong() method, as it always requires a long as input.

I think the check should be: 
HttpServletResponse.class.getMethod("setContentLengthLong", long.class)